vmdb/app/models/orchestration_stack_amazon.rb

```diff
@@ -1,13 +1,38 @@
 class OrchestrationStackAmazon < OrchestrationStack
+  def self.raw_create_stack(orchestration_manager, stack_name, template, options = {})
+    orchestration_manager.with_provider_connection(:service => "Orchestration") do |service|
+      service.stacks.create(stack_name, template.content, options).stack_id
+    end
+  rescue => err
+    _log.error "MIQ(#{self.class.name}##{__method__}) stack=[#{stack_name}], error: #{err}"
+    raise MiqException::MiqOrchestrationProvisionError, err.to_s, err.backtrace
+  end
+
   def raw_update_stack(options)
-    ext_management_system.with_provider_connection(:service => "CloudFormation") do |connection|
-      connection.stacks[name].update(options)
+    ext_management_system.with_provider_connection(:service => "CloudFormation") do |service|
+      service.stacks[name].update(options)
     end
+  rescue => err
+    _log.error "MIQ(#{self.class.name}##{__method__}) stack=[#{name}], error: #{err}"
+    raise MiqException::MiqOrchestrationUpdateError, err.to_s, err.backtrace
   end
 
   def raw_delete_stack
-    ext_management_system.with_provider_connection(:service => "CloudFormation") do |connection|
-      connection.stacks[name].delete
+    ext_management_system.with_provider_connection(:service => "CloudFormation") do |service|
```

Comment:
If the parent class had a `service_name`, would you be able to move `raw_delete`, and possibly `raw_update_stack` to the parent? Do you need raw version of these methods?  The subclass would just need to define these methods


```ruby
  attr_accessor :service_name

# or

  def service_name
    "CloudFormation"
  end
```